WSUCptSCapstone-Fall2022Spring2023/psd-gamifiedapp
# Sprint2/functional python ide

Closes #9 
Closes #24

Demo Video Link: https://youtu.be/X8nZtwSpGQg

**Overview:**

This pull request implements the primary level player UI, as well the the basic version of the Python IDE for the application. The player UI allow the user to play through two distinct phases, each with totally unique configurations. After succeeding on the first phase, the player advances to the second phase, loading all necessary configuration for the phase. Failures are met with specific responses that help the user to discover the correct solution.

**Details:**

This pull request features the addition of many files and changes throughout the code. In brief, the major changes are the following.

IronPythonContainer.cs: given Initialization priority and a separate timer variable for the "fast tick" speed

PhaseDefinition.cs: Renamed from LevelDefinition, given additional stored information about variable names, scripts, and starter code.

CRTShader.shader: Greatly simplified and replaced by built in postprocessing functionality

IDEController.cs: Expanded to handle and manage all of the UI in the IDE, including autogenerated fill in the blank sections.

LevelPlayerController.cs: Expanded to cover and manage all of the Level Player UI. Provides initialization and reinitialization utilities.

OutputController.cs: Manages the output window through which the AI gives feedback to the user.

PhaseUIController: Provides generic bindings for arbitrary phase UI

SequentialTyper.cs: Abstracts the setting of text in boxes to allow it to be dynamically typed one character a time with a slight delay.

In addition several text files for the new phase were added, as were plenty of configuration and automated project changes.

**Testing:**

The functionality for this pull request was tested using the following manual FT procedure.

1. Start Project
2. Confirm that a fill in the blank project with introductory text has loaded.
3. Write code that will not allow testVar to be reduced below -20,
4. Confirm failure and feedback given
5. Write code that will succeed
6. Confirm success
7. Confirm that new phase loads with new message, variables, and UI layout.
8. Complete this phase, confirm success.

The current build passes this process with all confirmed factors positive.
